Dockerfile

```diff
@@ -0,0 +1,33 @@
+# A base Docker image running Python 3.6.2 and Debian 8 "Jessie"
+# See: https://github.com/docker-library/python/blob/master/3.6/jessie/Dockerfile
+FROM python:3.6.2-jessie
+
+RUN apt-get update && \
+    apt-get install -y \
+    libboost-system-dev libboost-filesystem-dev libboost-chrono-dev \
+    libboost-program-options-dev libboost-test-dev libboost-thread-dev \
+    build-essential libtool autotools-dev automake pkg-config libssl-dev \
+    libevent-dev bsdmainutils \
+    libzmq3 miniupnpc
+
+# Install clang for potential use over g++.
+RUN apt-get install -y clang
```

Comment:
This should also be rolled up with the above apt install block, with the comment above it.

```docker
RUN apt-get update && apt-get install -y \
    libboost-system-dev \
    ...
    miniupnpc \
    # Install clang for potential use over g++.
    clang
```